### DISCUSSION TITLE
Can't get diff to work

### DISCUSSION THREAD

**Discussion by FrancisVila** [gitkraken/vscode-gitlens#1288] [Q&A]
Don't know whether this is a question or a bug report...
I want to diff between different branches or commits of the same file. 
when I do 
vscode > **git** > right click > **open changes (with diff tool)** I get 
`Unable to open changes in diff tool. No Git diff tool is configured (source: GitLens - git supercharged)`

I entered `git difftool --tool=winmerge `  then `git difftool --tool=araxis `  but the error persists.

Earlier I did manage to get a different error, in the OUTPUT section of the terminal: 
`ExternalDiffCommand TypeError: Cannot read property 'uri' of undefined`

Here's the version info:
- GitLens Version: v10.2.3
- Git Version: 2.27.0.windows.1

- VSCode Version: 
- 1.46.1
cd9ea6488829f560dc949a8b2fb789f3cdc05f5d
x64

**Comment by eamodio** [comment]
Do you want to compare the changes in an external difftool? Or do you want to see the diff directly in VS Code. If the latter, please use the _Open Changes with Revision..._ or _Open Changes with Branch or Tag..._ command as they will allow you to compare a file with different revisions directly within VS Code. If you wanted the former, then there is no built-in command to provide that ability. The _Open Changes (difftool)_ command opens the current revision in your configured Git difftool. If you don't have a configured difftool, see [here](https://stackoverflow.com/questions/6412516/configuring-diff-tool-with-gitconfig) to configure one.

**Reply by HernandoR** [reply]
Error: No diff tool found[2024-10-17 03:23:22.381] LocalGitProvider.openDirectoryCompare
Error: No diff tool found[2024-10-17 03:24:21.173] DiffFolderWithRevisionFromCommand
Error: No provider registered for 'output:extension-output-eamodio.gitlens-%231-GitLens'[2024-10-17 03:24:42.171] DiffFolderWithRevisionCommand
Error: No provider registered for 'output:extension-output-eamodio.gitlens-%231-GitLens'
